code/modules/mob/living/simple_animal/bees.dm

```diff
@@ -25,33 +28,55 @@
 		parent.owned_bee_swarms.Remove(src)
 	..()
 
+
+//Special death behaviour. When bees accumulate enough damage to 'die', they don't outright die.  Thus no call to parent
+//Instead the swarm strength (ie, size, or quantity of bees) drops and their health is refilled
+//Repeat until strength hits zero. only THEN do they die, and they qdel and leave no corpse in doing so
+//Because we don't have sprites for a carpet made of bee corpses.
+/mob/living/simple_animal/bee/death()
+	strength -= 1
+	if (strength <= 0)
+		if (prob(35))//probability to reduce spam
+			src.visible_message("\red The bee swarm completely dissipates.")
+		qdel(src)
+	else
+		health = maxHealth
+		if (prob(35))//probability to reduce spam
+			src.visible_message("\red The bee swarm starts to thin out a little.")
+
+	update_icons()
+
 /mob/living/simple_animal/bee/Life()
 	..()
 
 	if(stat == CONSCIOUS)
 		//if we're strong enough, sting some people
 		var/mob/living/carbon/human/M = target_mob
```

Comment:
Is it possible here that there is no target_mob, and could this runtime horribly? Or does the if-check on 57 cover that, I wonder?